include/extractor/guidance/mergable_road_detector.hpp

```diff
@@ -77,10 +77,15 @@ class MergableRoadDetector
     // situations with more than two roads that could participate in a merge
     bool IsDistinctFrom(const MergableRoadData &lhs, const MergableRoadData &rhs) const;
 
-  private:
     // check if two name ids can be seen as identical (in presence of refs/others)
     // in our case this translates into no name announcement in either direction (lhs->rhs and
     // rhs->lhs)
+    static bool HaveIdenticalNames(const NameID lhs,
```

Comment:
Instead of using a static class function, I would recommend making this a free function. Basically just add an own header file similar to `is_through_street.hpp`.